js_modules/dagit/packages/core/src/instance/InstanceConcurrency.tsx

```diff
@@ -0,0 +1,596 @@
+import {gql, useQuery, useMutation} from '@apollo/client';
+import {
+  PageHeader,
+  Heading,
+  Box,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Icon,
+  Menu,
+  MenuItem,
+  Mono,
+  Popover,
+  Spinner,
+  ButtonLink,
+  Table,
+  Tag,
+  TextInput,
+  Button,
+  NonIdealState,
+} from '@dagster-io/ui';
+import * as React from 'react';
+import {Redirect} from 'react-router-dom';
+
+import {showSharedToaster} from '../app/DomUtils';
+import {useFeatureFlags} from '../app/Flags';
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {useQueryRefreshAtInterval, FIFTEEN_SECONDS} from '../app/QueryRefresh';
+import {useTrackPageView} from '../app/analytics';
+import {useDocumentTitle} from '../hooks/useDocumentTitle';
+import {doneStatuses} from '../runs/RunStatuses';
+import {RunTable, RUN_TABLE_RUN_FRAGMENT} from '../runs/RunTable';
+import {RunTableRunFragment} from '../runs/types/RunTable.types';
+
+import {InstancePageContext} from './InstancePageContext';
+import {InstanceTabs} from './InstanceTabs';
+import {
+  InstanceConcurrencyLimitsQuery,
+  InstanceConcurrencyLimitsQueryVariables,
+  ConcurrencyLimitFragment,
+  SetConcurrencyLimitMutation,
+  SetConcurrencyLimitMutationVariables,
+  RunsForConcurrencyKeyQuery,
+  RunsForConcurrencyKeyQueryVariables,
+  FreeConcurrencySlotsForRunMutation,
+  FreeConcurrencySlotsForRunMutationVariables,
+} from './types/InstanceConcurrency.types';
+
+const RUNS_LIMIT = 25;
+
+const InstanceConcurrencyPage = React.memo(() => {
+  useTrackPageView();
+  useDocumentTitle('Configuration');
+  const {flagInstanceConcurrencyLimits} = useFeatureFlags();
+
+  const {pageTitle} = React.useContext(InstancePageContext);
+  const queryResult = useQuery<
+    InstanceConcurrencyLimitsQuery,
+    InstanceConcurrencyLimitsQueryVariables
+  >(INSTANCE_CONCURRENCY_LIMITS_QUERY, {
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const refreshState = useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
+  const {data} = queryResult;
+
+  if (!data) {
+    return (
+      <>
+        <PageHeader
+          title={<Heading>{pageTitle}</Heading>}
+          tabs={<InstanceTabs tab="concurrency" refreshState={refreshState} />}
+        />
+        <Box padding={{vertical: 64}}>
+          <Spinner purpose="section" />
+        </Box>
+      </>
+    );
+  }
+
+  const limits = data.instance.concurrencyLimits;
+  return (
+    <>
```

Comment:
You may want to wrap this in a `<Page>` component to ensure that the page is scrollabe if it goes past the bottom of the viewport.